feat: New Footer

tofumatt opened the pull request:
Fix #2518.

Simple style changes to the footer that brings it in line with the new style.

### Desktop

<img width="1028" alt="screenshot 2017-06-06 18 59 20" src="https://user-images.githubusercontent.com/90871/26844134-57ce6268-4aea-11e7-9de2-11c8873f71f5.png">
<img width="982" alt="screenshot 2017-06-06 18 58 07" src="https://user-images.githubusercontent.com/90871/26844135-57d42266-4aea-11e7-8e6d-49fd94db7fa1.png">

### Mobile

<img width="346" alt="screenshot 2017-06-06 18 59 12" src="https://user-images.githubusercontent.com/90871/26844143-5bdf682a-4aea-11e7-8096-205bfad34e64.png">
<img width="355" alt="screenshot 2017-06-06 18 58 12" src="https://user-images.githubusercontent.com/90871/26844144-5bdfaf60-4aea-11e7-8c0b-01480fae8aec.png">

tofumatt commented on src/amo/components/Footer/index.js:
```diff
@@ -0,0 +1,79 @@
+
+        <a
+          className="Footer-mozilla-link"
+          href="https://mozilla.org/"
```
I guess we could add lang to this as well; I'll do that.

tofumatt commented on src/amo/components/Footer/styles.scss:
```diff
@@ -0,0 +1,49 @@
+}
+
+.Footer-language-and-links {
+  max-width: 220px;
```
This is pretty protective against screens that are lower than `220px`... realistically we could probably just make this `width: 220px` as that's what's in the desktop mocks and fits on even phones smaller than `280px` wide. Let me know.

tofumatt commented on src/amo/components/LanguagePicker.scss:
```diff
@@ -9,25 +8,26 @@ $selector-padding: 10px;
   line-height: 1;
-  padding: $selector-padding $selector-padding #{$selector-padding - 1} #{$selector-padding * 3};
+  max-width: 220px;
+  padding: 5px 15px 3.5px 30px;
```
Could probably use a mixin for this; I'll see if we have one.

muffinresearch commented on src/amo/components/Footer/index.js:
```diff
@@ -0,0 +1,79 @@
+            ref={(ref) => { this.languagePicker = ref; }} />
+          <ul className="Footer-links">
+            <li>
+              <a href={`https://www.mozilla.org/${lang}/privacy/websites/`}
```
Seeing the lang made me wonder - does mozilla.org handle it gracefully if we get them a lang they don't cater for?

tofumatt replied:
There seem to be certain locales it will redirect to if it doesn't have one available, but they need to be known (so `en-CA` redirects to `en-US`). I'll see about all of them, but given it will 404 if it doesn't redirect, we could just omit the locale to be safe. moz.org should figure out the user's locale like we do so that'll only an annoying experience if you manually set your locale on AMO to be different than the one assigned to you.

muffinresearch replied:
Yeah it's a shame that sometimes it falls apart. I know MDN in the past suffered from problems if the locale was one it didn't know.

muffinresearch commented on src/amo/components/Footer/index.js:
```diff
@@ -0,0 +1,79 @@
+          title={homepageText}
+        >
+          <Icon className="Footer-mozilla-logo" name="mozilla" />
+          <span className="visually-hidden">{homepageText}</span>
```
Can the Icon have an alt attribute and then this span would not be needed?

tofumatt replied:
I think I'd need to add it but that's a good idea.

tofumatt commented:
Okay I added the alt text support to `Icon` and removed the lang from the footer URLs as they had the potential to route to 404s. Ready for another r?

tofumatt commented:
Shoot, asked for a review and forgot to force push because I rebased this onto master. It's all set for r? now

muffinresearch commented on src/ui/components/Icon/index.js:
```diff
@@ -4,18 +4,29 @@ import classNames from 'classnames';
+    }
+
+    return (
+      <i className={classNames('Icon', `Icon-${name}`, className)} {...props} />
```
Ah I didn't realise this compont use used an `i` element under the covers. This makes the alt attribute a little bit weird but I suppose the alt convention works even if it's not actually an alt attribute on an img.
